### ISSUE TITLE
Cli export / import roles including permissions Part 2 (refinement)

### ISSUE THREAD

**Issue by gbonazzoli** [apache/airflow#36580]
### Apache Airflow version

main (development)

### What happened?

This is the refinement of: https://github.com/apache/airflow/issues/36128

The CLI command `airflow roles export` now in the main branch exports roles along with their associated permissions.

Good !

There is only a little aesthetic inaccuracy as you can see form the airflow roles export output:

```
airflow@airflow-kube-worker-848886d894-gv4wn:~$ airflow roles export b.json
38 roles with permissions successfully exported to b.json
```
it says 38 roles, actually they are only 4 roles but the permissions associated have multiplied them.

I looked at the code and I think we have the same problem also in the import phase.

### What you think should happen instead?

My suggestion is that the command should print out: `4 roles exported with 38 linked/granted permissions`.

`linked` in export and `granted` in import

### How to reproduce

gives the command above.

### Operating System

branch main

### Versions of Apache Airflow Providers

```
apache-airflow                           2.8.0
apache-airflow-providers-amazon          8.13.0
apache-airflow-providers-celery          3.5.0
apache-airflow-providers-cncf-kubernetes 7.11.0
apache-airflow-providers-common-io       1.1.0
apache-airflow-providers-common-sql      1.9.0
apache-airflow-providers-docker          3.8.2
apache-airflow-providers-elasticsearch   5.3.0
apache-airflow-providers-ftp             3.7.0
apache-airflow-providers-google          10.12.0
apache-airflow-providers-grpc            3.4.0
apache-airflow-providers-hashicorp       3.6.0
apache-airflow-providers-http            4.8.0
apache-airflow-providers-imap            3.5.0
apache-airflow-providers-microsoft-azure 8.4.0
apache-airflow-providers-mysql           5.5.0
apache-airflow-providers-odbc            4.2.0
apache-airflow-providers-openlineage     1.3.0
apache-airflow-providers-postgres        5.9.0
apache-airflow-providers-redis           3.5.0
apache-airflow-providers-salesforce      5.6.0
apache-airflow-providers-sendgrid        3.4.0
apache-airflow-providers-sftp            4.8.0
apache-airflow-providers-slack           8.5.0
apache-airflow-providers-snowflake       5.2.0
apache-airflow-providers-sqlite          3.6.0
apache-airflow-providers-ssh             3.9.0
google-cloud-orchestration-airflow       1.10.0
```

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

kubernetes v1.28.4

helm v3.13.1

### Anything else?

kubernetes v1.28.4

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by gbonazzoli** [comment]
@potiuk 

I need one simple information.

When the new code is approved and the issue is closed, 
how can you be sure that the code goes out into an airflow release ?

Where is the join between release and new code ?

In more practical way, will it be included in Airflow 4.8.2 or 4.9.0 ?

Thanks

**Comment by gbonazzoli** [comment]
Perfect, we will have a check on 2.9.x

**Comment by sandeyshc** [comment]
Has this been released in any version of Airflow? I couldn't find this issue mentioned in any of the Airflow release documentation.

**Comment by gbonazzoli** [comment]
@sandeyshc 

For sure it is present on Airflow 2.10.x so I suppose that was released on Airflow 2.9.0

**Comment by sandeyshc** [comment]
Airflow release notes haven't been updated with the details, so I wanted to verify which version it is. Thank you for the information.
